src/rust/runtime/memory/mod.rs

```diff
@@ -85,45 +107,18 @@ pub trait MemoryRuntime {
 
     /// Releases a scatter-gather array.
     fn sgafree(&self, sga: demi_sgarray_t) -> Result<(), Fail> {
-        // Check arguments.
-        // TODO: Drop this check once we support scatter-gather arrays with multiple segments.
-        if sga.sga_numsegs != 1 {
-            return Err(Fail::new(libc::EINVAL, "demi_sgarray_t has invalid segment count"));
-        }
-
-        if sga.sga_buf == ptr::null_mut() {
-            return Err(Fail::new(libc::EINVAL, "demi_sgarray_t has invalid DemiBuffer token"));
+        match self.into_buf(&sga) {
```

Comment:
Please do not remove TODO comments. For example, we still do not support multiple sga segments (I think), so we probably shouldn't remove it.

Reply:
Actually, the TODOs are moved into `into_buf` function.

Reply:
Ah ok, all good then!